### PR TITLE
khard: add option to contact module for khard dir

### LINKED ISSUE #4531: bug: Incompatibility between vdirsyncer and khard modules
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

I have enabled both khard and vdirsyncer programs. The idea is that vdirsyncer fetches&uploads the contacts that are listed&edited with khard.

It seems that it isn't possible to have both vdirsyncer and khard configured together with home-manager because:
1. vdirsyncer creates a subfolder within `accounts.contact.accounts.<name>.local.path` and puts the vcf files in the subfolder.
2. khard reads the vcf files from `accounts.contact.accounts.<name>.local.path` directly and doesn't look recursively inside the subfolders.
As a result, khard finds no vcf files and thus lists no contacts.

Said differently, here is part of my vdirsyncer configuration:

```ini
[storage contacts_nico_local]
fileext = ".vcf"
path = "~/configuration/contacts"
```

and here is part of my *working* hand-written khard configuration:

```ini
[addressbooks]
[[contacts]]
path = ~/configuration/contacts/172ce...
```

As you can see, the configuration file references the subfolder `172ce.../` instead of directly the path configured for vdirsyncer.

Here is an excerpt of what home-manager generates for khard:

```ini
[addressbooks]
[[nico]]
path = ~/configuration/contacts
```

This doesn't work because of the missing subfolder.

Here is the home-manager configuration that I cannot use:

```nix
{ pkgs, lib, ... }:

{
  programs.vdirsyncer = { enable = true; };

  programs.khard = {
    enable = true;
    settings = {
      ...
    };
  };

  accounts.contact.accounts = {
    nico = {
      vdirsyncer = {
        enable = true;
        ...
      };
      khard = { enable = true; };
      local = {
        path = "~/configuration/contacts";
        ...
      };
      remote = { ... };
    };
  };
}
```

### Maintainer CC

@olmokramer @antonmosich @teto @ayyess @loewenheim @jorsn 

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.5.5-200.fc38.x86_64, Fedora Linux, 38 (Workstation Edition), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixpkgs"`
 - channels(temp): `"home-manager, nixpxkgs"`
 - channels(cassou): `"home-manager-22.11.tar.gz, nixpkgs"`
 - nixpkgs: `/home/cassou/Documents/projects/nix-system/nixpkgs`
```

### PR DESCRIPTION
**Pull request by octvs**
- Add an extra option, `accounts.contact.accounts.<name>.khard.defaultCollection` to hardcode a subdir for khard to function as intended.
  - Khard expects to be given a collection directory directly, from which there can be multiple of in a vdir. This contrasts khal or vdirsyncer which support recursive search.

Fixes #4531

### Description

Details can be found on the commit msg. This is my first time trying to contribute, thus I appreciate any feedback & shown patience.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- ~~If this PR adds a new module~~

   - ~~[ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).~~

#### Maintainer CC

@olmokramer @antonmosich

### CODE REVIEW

**Comment by octvs** [comment]
> Test cases updated/added. 
I wasn't sure if this is applicable for this minor tweak, but I can look into it.

File: modules/programs/khard.nix [hunk 1]
```diff
@@ -79,7 +84,12 @@ in {
       [addressbooks]
       ${lib.concatMapStringsSep "\n" (acc: ''
         [[${acc.name}]]
-        path = ${acc.local.path}
+        path = ${
+          lib.strings.concatStringsSep "/" [
```
Comment (sumnerevans) [inline modules/programs/khard.nix:88]:
```suggestion
          lib.path.subpath.join [
```

File: modules/programs/khard.nix [hunk 2]
```diff
@@ -68,6 +68,11 @@ in {
       type = with lib.types;
         attrsOf (submodule {
           options.khard.enable = lib.mkEnableOption "khard access";
+          options.khard.defaultCollection = lib.mkOption {
+            type = types.str;
+            default = "";
+            description = "Sub-vdir to hardcode for khard";
```
Comment (sumnerevans) [inline modules/programs/khard.nix:74]: I don't quite understand this, but I think this is more descriptive of what it is.

```suggestion
            description = "Sub-vdir where the khard address book resides.";
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/programs/khard.nix (hunk 2 above)
Comment (octvs) [reply modules/programs/khard.nix]: Thank you very much for the review!
I accidentally destroyed the src branch for the PR, but I've already took these into account, could we continue on #5220?

**Pull request closed without merging**